src/azure.ad/types_error.bal

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+const AD_CLIENT_ERROR = "{ballerinax/azure.ad}AdClientError";
+
+public type AdClientError error<string, Details>;
```

Comment:
```suggestion
public type AdClientError error<AD_CLIENT_ERROR, Details>;
```

Reply:
We need to leave it as <string, Details>

The Graph API errors return a code and that code is used for the `reason` for those errors.